tools/libSwiftSyntaxParser/CMakeLists.txt

```diff
@@ -14,7 +14,8 @@ if(NOT SWIFT_BUILT_STANDALONE AND NOT CMAKE_C_COMPILER_ID MATCHES Clang)
   add_dependencies(libSwiftSyntaxParser clang)
 endif()
 target_link_libraries(libSwiftSyntaxParser PRIVATE
-  swiftParse)
+  swiftParse
+  swiftLocalization)
```

Comment:
I was getting linker error message when building `libSwiftSyntaxParser` and `swift-dependency-tool`. I thought that was because we're not linking `swiftLocalization` here. 

After doing that change it worked. So I have two questions:
1. Is it the correct way to fix it?
2. I'm not entirely sure why it is  happening now and not before? I.e. why my change to call `setLocalization` from `DiagnosticEngine::diagnosticMessageFor` caused that linker error?

An example of the linker error:
```
[1/2] Linking CXX shared library lib/lib_InternalSwiftSyntaxParser.dylib
FAILED: lib/lib_InternalSwiftSyntaxParser.dylib
Undefined symbols for architecture x86_64:
...
...
...
"swift::diag::YAMLLocalizationProducer::YAMLLocalizationProducer(llvm::StringRef, bool)", referenced from:
      swift::DiagnosticEngine::setLocalization(llvm::StringRef, llvm::StringRef) in libswiftAST.a(DiagnosticEngine.cpp.o)
  "swift::diag::SerializedLocalizationProducer::SerializedLocalizationProducer(std::__1::unique_ptr<llvm::MemoryBuffer, std::__1::default_delete<llvm::MemoryBuffer> >, bool)", referenced from:
      swift::DiagnosticEngine::setLocalization(llvm::StringRef, llvm::StringRef) in libswiftAST.a(DiagnosticEngine.cpp.o)
ld: symbol(s) not found for architecture x86_64
```

Reply:
It used be to called from different module before - Frontend.cpp belongs to `swiftFrontend`, I think it should stay that way.

Reply:
Makes sense, thank you!